### PR TITLE
Don't prompt for sign in after user has ignored once. 

### LINKED ISSUE #1389: Everytime I open VSCode I get a notification that I need to sign in to use the extension
We should serve as a good role model here and not abuse notifications:

image

Please either add a button to never notify again or simply not show it.

//cc @RMacfarlane

### PR DESCRIPTION
**Pull request by JacksonKearl**
Fixes #1389.

Added a new persistent global state store, as it seems like the existing solution (authentication/keychain.ts) was pretty heavily tied to passwords/keytar.

### CODE REVIEW

File: src/github/credentials.ts [hunk 1]
```diff
@@ -13,14 +13,17 @@ import { GitHubServer } from '../authentication/githubServer';
...
 import Logger from '../common/logger';
+import * as PersistentState from '../common/persistentState';
 import { handler as uriHandler } from '../common/uri';
 import { createHttpLink } from 'apollo-link-http';
 import fetch from 'node-fetch';
 import { ITelemetry } from '../common/telemetry';
 
 const TRY_AGAIN = 'Try again?';
 const SIGNIN_COMMAND = 'Sign in';
+const IGNORE_COMMAND = 'Close';
```
Comment (RMacfarlane) [inline src/github/credentials.ts]: Let's make this "Don't show again" for clarity

File: src/common/persistentState.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as vscode from 'vscode';
```
[the remark this answers is not in the document]
Comment (JacksonKearl) [reply src/common/persistentState.ts]: Okay. Lots of files don't have it, should we go and add it everywhere too?

File: src/common/persistentState.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+export const MISSING = {} as const;
+
+export function init(ctx: GlobalStateContext) {
+	defaultStorage = ctx.globalState;
+};
+
+export const fetch = (scope: string, key: string): unknown => {
+	if (!defaultStorage) {
+		throw new Error('Persistent store not initialized.')
```
[the remark this answers is not in the document]
Comment (JacksonKearl) [reply src/common/persistentState.ts]: Fixed

File: src/common/persistentState.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/common/persistentState.ts]: I think other files that don't have it were added by GitHub folks, we're maybe supposed to also add the header when we modify those files? @rebornix do you know?

For now I would just add it to this one

File: src/github/credentials.ts (hunk 1 above)
Comment (JacksonKearl) [reply src/github/credentials.ts]: Done!

File: src/common/persistentState.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JacksonKearl) [reply src/common/persistentState.ts]: Got it, done

[the remark this answers is not in the document]
Comment (rebornix) [reply src/common/persistentState.ts]: @RMacfarlane you are right, those files were contributed by GitHub folks so we didn't force the copyright headers to be there. We can do that when modifying them.

**Comment by rebornix** [comment]
@JacksonKearl since this line 
https://github.com/microsoft/vscode-pull-request-github/pull/1406/files#diff-a46298735cf7c583b6e833c09a3653e1R118 is commented out, I don't think ignoring prompt actually works, can you help confirm that?

**Comment by JacksonKearl** [comment]
Oops, yeah I had that in for testing. Fixed on master.

**Pull request merged**